sklearn/feature_selection/_univariate_selection.py

```diff
@@ -206,6 +206,10 @@ def chi2(X, y):
     contain only **non-negative features** such as booleans or frequencies
```

Comment:
Maybe we can take the opportunity to improve this part as well while we are at it. Because to be honest I don't find it very clear ...

@glemaitre I have seen you have been involved in chi2 issues in the past so maybe you have an informed opinion on this:

Maybe something like:
```
contain only **non-negative categorical features**
```
or
```
contain only **non-negative integer features**
```